src/QirRuntime/test/QIR-static/qir-test-ouput.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include <iostream>
+#include <sstream>
+
+#include "catch.hpp"
+
+#include "qirTypes.hpp"
+#include "quantum__qis_internal.hpp"
+
+extern "C" void Microsoft__Quantum__Testing__QIR__Out__MessageTest__body(QirString*);        // NOLINT
```

Comment:
Would it help to add a comment that we are aware of the signature mismatch between this decl and the generated function in QIR but it's safe in this case?

Reply:
Hm... Very good question... I didn't even notice that (strictly speaking) the parameter type is different between here and in the generated `.ll`. And such a function call (strictly speaking) may be unsafe.
To stay safe, we can do the following:
* assume that `%String*` and `%struct.QirString*` are the same as (are just aliases of) `void*`;
* for that we can replace the current type definitions to `%String = type void`, `%struct.QirString = type void`, or, even simpler, use `void*` everywhere instead of `%String*` and `%struct.QirString*`.

This will let us avoid all those casts (in `.ll` files) between `%String*` and `%struct.QirString*`, and to stay safe when making calls between `.cpp` and `.ll` files.
What do you think?

For now in this line I'm changing the parameter type to `void*` for the call to be safe.